### ISSUE TITLE
Inline Math No Longer Showing

### ISSUE THREAD

**Issue by acircleda** [rstudio/rstudio#7947]
### System details

    RStudio Edition : Desktop
    RStudio Version : Version 1.4.869
    OS Version      :  Windows 10
    R Version       :  4.0.2

### Steps to reproduce the problem

In any R Markdown document, the following code will not display correctly for me:

```
This is a Greek letter $\sigma^2$

This is regular text $p<.05$
```

### Describe the problem in detail

In non, visual editor mode, the inline math text which previously worked still turns orange but a pop-up preview is no longer displayed.
In visual editor mode, the inline math simply does not show.

![Non Visual Editor Mode](https://user-images.githubusercontent.com/51770624/94921921-3f0f5e80-0487-11eb-9038-fb1da91c4d5b.png)

![Visual Editor Mode](https://user-images.githubusercontent.com/51770624/94921945-4afb2080-0487-11eb-8638-88b3124d2e9a.png)

### Describe the behavior you expected

Both should as inline math

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@acircleda Thank you for raising the issue! We'll review this as we continue development of RStudio.

I can reproduce the issue on MacOS 11 Big Sur beta 8 in RStudio Desktop 1.4.869. Note, though, that the correct behavior occurs when clicking where the text should be—in this case, to the right of "letter" and "text".

This is not a problem in RStudio Server 1.4.869 on Red Hat 8.2 or 1.4.765 on Ubuntu 20.04.

**Comment by adamconroy** [comment]
Hi @acircleda, can I get a little more information from you as I am not currently able to reproduce your issue.

1. Does this apply for all themes, both dark and light, for you?
2. Can you open up your debug console (right click -> inspect and click on Console) and tell me if there are any error messages. I'm wondering if you might not be able to load the MathJax font and if that is the cause. It would explain why I can't reproduce this in my environment because I have the fonts installed into the operating system.

Thanks!

**Comment by acircleda** [comment]
Hi. I can confirm it was happening with light and dark themes (my default is the Cobalt theme). I changed my font from Courier New to Consolas and text is appearing again. **However**, when viewing the math in the new visual editor, I can't seem to click to the end of the line to continue writing. If I *click* it brings me back to the text between `$$`. However, if I use my arrow keys I can continue the line.

**Comment by adamconroy** [comment]
@acircleda are you able to open your debug console as I described above? I have a feeling that it's not loading the specific MathJax font. This has nothing to do with your actual editor font, the math is drawn using a special font not under your control and it might be missing for you.

**Comment by adamconroy** [comment]
@acircleda If you've updated to the latest daily can you let us know if this is still happening to you. If you can open your dev console with right click -> inspect, I'm interested if you're seeing any errors there. I am not able to reproduce this in the various environments I've tried but I suspect it's an external loading issue in other environments.

**Comment by acircleda** [comment]
Hi, sorry for the late reply. 
I'm on Version 1.4.869.
After loading a visual editor file, here is what the errors are:

﻿```
Failed to load resource: the server responded with a status of 404 (Not Found) :27741/ace/snippets/yaml.js:1 
Failed to load resource: the server responded with a status of 404 (Not Found): MathJax_Math-Italic.woff:1 
Failed to load resource: the server responded with a status of 404 (Not Found): MathJax_Main-Regular.woff:1 
```

**Comment by kevinushey** [comment]
I'm also not able to reproduce (1.4.992 on Windows 10). I see the same warnings as reported by @acircleda in the console, but those ultimately seem harmless.

**Comment by adamconroy** [comment]
I've tried this a few different ways, both on Server and Desktop from clean installs with different Latex formulas and cannot reproduce.  I'm going to close this for now but keep an eye out for other reports. 

@acircleda are you able to get access to a clean Windows machine and try the latest daily there? Can you try specifically in the Clouds Midnight theme as well (the primary one that I've been testing in)? I'm wondering if the cause is some conflicting font installed on your machine that's corrupting it. You can also try `inspecting` from the dev console (click the box where the arrow in the picture is coming from or right click -> inspect) the Latex element to see if any clues can come from that. Paying close attention to what I've highlighted in the attached picture.

![pDwppoS](https://user-images.githubusercontent.com/136863/98163024-2e03a400-1eb0-11eb-8b03-426f5415b881.png)

**Comment by kfeinauer** [comment]
@adamconroy I also reproduced this with the given text:

```
This is a Greek letter $\sigma^2$

This is regular text $p<.05$
```

The problem reproduced using 1.4.1027 on OSX Catalina. Interestingly enough, it seems sometimes the popup preview shows in regular editor mode, and sometimes it doesn't. Regardless, the symbols are never displayed in the Visual Editor.

**Comment by kfeinauer** [comment]
![Nov-13-2020 15-07-58.gif](https://codetreeusercontent.s3.amazonaws.com/uploads/be733063-be88-49bc-8573-9ffc3564e1da/Nov-13-2020+15-07-58.gif)

**Comment by kfeinauer** [comment]
![Nov-13-2020 15-07-58](https://user-images.githubusercontent.com/28230621/99121592-48313680-25c2-11eb-9e8a-dcf92d492550.gif)

**Comment by adamconroy** [comment]
I'll try on my Mac. I've tried a bunch of different ways and cannot reproduce on my Windows machine.

**Comment by adamconroy** [comment]
With the latest daily on Mojave I cannot reproduce. The bug was filed on Windows so that seems unlikely to be the issue. I wonder why it seems fine for me and not anyone else, across multiple of my machines.

<img width="310" alt="Screen Shot 2020-11-13 at 4 42 36 PM" src="https://user-images.githubusercontent.com/136863/99124082-7c5f2400-25cf-11eb-9cfa-a3dd6a5db857.png">


@jmcphers do you have any ideas?

**Comment by kfeinauer** [comment]
Could this be a themeing issue? My theme is standard.

**Comment by adamconroy** [comment]
I can't reproduce it in any theme, just tried. If you inspect where the text is supposed to be when you're getting the bug what do you see? It just being erroneously set to white text was my initial thought above but without being able to repro it I haven't been able to look deeper into it.

**Comment by kfeinauer** [comment]
![Screen Shot 2020-11-13 at 4 06 24 PM](https://user-images.githubusercontent.com/28230621/99125675-305db080-25ca-11eb-83a0-8428b1398971.png)

**Comment by adamconroy** [comment]
Sorry, this is tough to ask for all the details here. What I'm curious about is the final calculated style. You'll see that on the side. What is the text color? What is the font set to? etc. If you look at the image I posted above doing the inspection the information on the right side is the most important. I'm really just curious if the font color is somehow being set to white but that it's actually there.

**Comment by adamconroy** [comment]
And to add on to that, if the text is actually there...is the calculated style somehow not setting the font correctly (and is the font actually being loaded). I'd be down to do a call on Monday if you wanted to help debug this a little bit more. Unfortunately this is really tough for me to figure out without being able to repro.

**Comment by kfeinauer** [comment]
@adamconroy It's now working for me in the latest daily, 1.4.1032.

@acircleda Can you verify that this is now fixed for you on the same version (or latest daily).

**Comment by mikebessuille** [comment]
@ronblum  would you mind trying to repro this again, since you were able to repro previously?  If you can't, we will close this.

**Comment by kfeinauer** [comment]
Was able to reproduce again @adamconroy:

Steps to reproduce (on OSX):
1. Open an Rmd that has the math that we've been testing above. Make sure you open without rendering visual mode (regular source mode).
2. Click on the math formatting, and you'll see the render popup as expected.
3.  Select the line `This is a Greek letter`, including the math in the selection
4. However your cursor over the sigma and verify that the popup occurs
5. While keeping your selection, move your mouse to hover over the rho greek letter in the `This is regular text` line
6. Notice that the render popup changes to show the rho
7. Now move your mouse cursor back to sigma and hover over it
8. The render popup does not change (still shows rho), and as soon as you click off your selection to dismiss it, you can never get it to show again for either rho or sigma.
9. Toggle Visual mode, and note that the math is not shown inline as it should be.

Note that I am testing with the Ambiance theme.

If you look at my divs, you'll notice I am missing a lot of the actual math subelements that you have when it's working.

![Screen Shot 2020-11-20 at 10 14 48 AM](https://user-images.githubusercontent.com/28230621/99822834-4cf16f80-2b19-11eb-998a-1b7ffc44f963.png)

Spoke with @ronblum and he was also able to reproduce in this manner (on Windows):

> When I start RStudio with an Rmd open in source mode or visual mode, it works.
> If I start in visual mode and switch back and forth, it stop working in source mode, but usually continues to work in visual mode.
> If I’m in source mode and 1) save the file, 2) switch to a different file and back, or 3) switch between source and visual mode (this is not consistent), it’ll stop working for any file until I restart.
> If I’m visual mode, sometimes the image starts off correctly. But when I click on the image, the code is revealed, but then when clicking elsewhere, the image is gone.

**Comment by kfeinauer** [comment]
@adamconroy Was able to simplify the repro even further: there's no need to select any text. Simply:

1. Hover over Sigma in source mode
2. Wait for redner popup
3. Then immediately hover over rho
4. Wait for render popup
5. Then hover back to sigma and it will not change from rho. Once the render popup goes away after clicking the mouse, you're left in the broken state.

**Comment by kfeinauer** [comment]
Verified fixed 1.4.1066
